packages/firebase/useAuth/index.ts

```diff
@@ -0,0 +1,22 @@
+import { computed, ComputedRef, ref, Ref } from 'vue'
+import firebase from 'firebase'
+import 'firebase/auth'
+
+export interface FirebaseAuthOptions {
+  isAuthenticated: ComputedRef<boolean>
+  user: Ref<firebase.User | null>
+}
+
+export function useAuth() {
+  const { auth } = firebase
```

Comment:
Is that safe to use global instance? Or should we at least provide an option to override it?

Reply:
What about something like this snippet below? I ran into an issue earlier where I was calling `firebase.init` in the same component that I was calling useAuth, but as long as they call `firebase.init` in something like `main.js` before `useAuth` is used, it shouldn't be a problem. I guess maybe the only problem might be if they use the `useAuth` in the main.js file then it could become a problem The snippet below would also allow the user to provide a non-global instance of firebase auth, but if they don't it will just fallback to the global instance.

```ts
export function useAuth(authInstance?: typeof firebase.auth) {
  const auth = authInstance ? authInstance() : firebase.auth()

  const user = ref<firebase.User | null>(auth.currentUser)
  const isAuthenticated = computed(() => !!user.value)

  auth.onIdTokenChanged(authUser => user.value = authUser)

  return {
    isAuthenticated,
    user,
  }
}
```

Reply:
Yeah, that's much better. But instance of authInstance, it's probably better to be authConstructor? Also think it might be possible to support both constructor and instance?

Reply:
Oh so you mean something like supporting the use of both `useAuth(firebase.auth)` and `useAuth(firebase.auth())`?

If that is what you mean then I'll update make an update tomorrow to support that :)

Reply:
Exactly. Thanks.

Reply:
Sorry it took a while, got busy with some stuff. Just implemented it though, let me know what you think 😅